MaxText/train.py

```diff
@@ -154,6 +155,11 @@ def write_metrics_to_tensorboard(writer, metrics, step, config):
 
 def save_checkpoint(checkpoint_manager, step, state, dataset_type="c4", data_iterator=None):
   """Wrapper for saving checkpoint"""
+  if isinstance(checkpoint_manager, emergency_checkpoint_manager.CheckpointManager):
```

Comment:
Can we not use emergency checkpointing with regular checkpointing? It looks to me like the emergency checkpoint manager is missing some of the key features of the regular one - (persistenet storage/GCS for the `period` saves, async option), but am still not sure what an emergency checkpoint is

Reply:
The saving is by default async. And as explained above, ECM has the regular checkpoint manager as one of its components.

Reply:
Also, ECM.save does not support composite args for now, so treating the case separately upfront.